azure/datalake/store/lib.py

```diff
@@ -86,6 +90,8 @@ def auth(tenant_id=default_tenant, username=default_username,
         the secret associated with the client_id
     resource : str
         resource for auth (e.g., https://management.core.windows.net/)
+    require_2fa : bool
+        indicates this authentication attempt requires two-factor authentication
```

Comment:
By wrapping ADAL instead of just ask a token, you will come back later here when any changes is applied to ADAL to change your method and add parameters.
Please consider that ADAL will be replaced in 2017 by MSAL (a mew lib with token caching across SDK languages). You approach seems dangerous to me.